sormas-app/app/src/main/java/de/symeda/sormas/app/backend/disease/DiseaseConfigurationDao.java

```diff
@@ -43,6 +50,38 @@ public String getTableName() {
 		return DiseaseConfiguration.TABLE_NAME;
 	}
 
+	public List<Disease> getAllDiseases(Boolean active, Boolean primary, Boolean caseBased) {
+
+		final Set<Disease> diseases = EnumSet.noneOf(Disease.class);
+
+		final List<DiseaseConfiguration> diseaseConfigurations = queryForAll();
+
+		final User currentUser = ConfigProvider.getUser();
+
+		for (DiseaseConfiguration configuration : diseaseConfigurations) {
+			final Disease limitedDisease = currentUser.getLimitedDisease();
+			if (limitedDisease == null) {
+				final Disease disease = configuration.getDisease();
+				boolean activeFlagMatches = active.equals(configuration.getActive()) || active.equals(disease.isDefaultActive());
```

Comment:
Does this make sense? Won't this return a disease as active that is specifically set to inactive on the server, but is active by default?